aips/code-verification-api.md

```diff
@@ -0,0 +1,126 @@
+
+Of course, code verification can be performed locally through the SDK. However, to increase accessibility and usability, it's most desirable to provide it through an API in the explorer.
+
+In this proposal, I aim to address these issues by introducing a smart contract code verification API. This API will assist developers in confirming the code with which they have deployed their smart contracts. This will enhance the safety and reliability of the code and enable potential problems to be discovered and corrected in advance.
+
+## Impact
+
+This change will impact smart contract developers and users of the Aptos network. Developers will be able to use this API to verify their code, and users can gain confidence in the safety of the verified smart contracts.
+
+When users know that a smart contract has been verified through this API, they gain trust that they can use that smart contract more safely. This enables users to engage in transactions and interactions within the network with greater confidence.
+
+Therefore, this change will contribute to enhancing the overall stability and trustworthiness of the Aptos network.
+
+## Rationale
+
+This proposal is the most effective way to improve the stability and trustworthiness of the Aptos network. This API will help developers make smart contracts safer and play a crucial role in enhancing the network's transparency and increasing user trust. Since Aptos is designed to store the code and the compiled bytecode on-chain when deploying Move code, verification is more convenient compared to other chains.
+
+## Specification
+
+### **Request**
+
+For verification of the smart contract code, developers must send a **`POST`** request to the API endpoint. It must include the following fields:
+
+- **`address`**: The address where the smart contract to be verified is deployed.
+- **`moduleName`**: The name of the module to be verified.
+- **`compilerVersion`**: The version of the Move compiler used for compilation.
+- **`network`**: The name of the network where the smart contract is deployed.
+
+### **Response**
+
+The API returns the verification result in JSON format. The response body includes the following fields:
+
+- **`address`**: The address of the verified smart contract.
+- **`network`**: The name of the network where the verification took place.
+- **`isMatched`**: Whether the smart contract code matches the deployed bytecode.
+- **`module`**: The name of the verified module.
+- **`package`**: The name of the verified package.
+- **`txHash`**: The hash value of the smart contract deployment transaction.
+
+## Operation Details(example)
+
+- Users request verification for a specific module in the explorer.
+- When making a request, parameters such as address, network, module name, and compiler version are sent together.
+- The API server compiles the on-chain metadata based on user information and compares it to the on-chain bytecode.
```

Comment:
Perhaps  'The API server identifies the on-chain package metadata which declares the module and compiles this package. The resulting bytecode is compared with the bytecode on-chain.'

In any case its important to understand that we can only compile packages as a whole, because the settings are all in the Move.toml. Perhaps this API should really be on package level not module level. We need to compile all of it anyway.